### PR TITLE
command: add TF_CLI_ARGS to specify additional CLI args

### PR DESCRIPTION
**Pull request by mitchellh**
This introduces an env var `TF_CLI_ARGS` that can be used to specify additional CLI args. 

The primary use for this is easier CI and automation integration.

The flag is parsed like a shell line, so the value `TF_CLI_ARGS="-foo 'bar baz'"` will turn into `["-foo", "bar baz"]` internally. 

The additional args are appended _before_ any args specified on the CLI _after_ the command. So, if you invoke `TF_CLI_ARGS="-force" terraform apply -input=false` then it is equivalent to specifying `terraform apply -force -input=false`. This ordering forces the CLI args to take precedent over the env var.

Updating the docs now.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -129,9 +135,45 @@ func wrappedMain() int {
...
+				"Error parsing extra CLI args from %s: %s",
+				EnvCLI, err))
+			return 1
+		}
+
+		// Find the index to place the flags. We put them exactly
+		// after the first non-flag arg.
+		idx := -1
+		for i, v := range args {
+			if v[0] != '-' {
```
Comment (jbardin) [inline main.go]: An arg can be 0 length -- even more likely in an automated system like TFE. 
`strings.HasPrefix`?

File: main.go [hunk 2]
```diff
@@ -129,9 +135,45 @@ func wrappedMain() int {
...
+			}
+		}
+
+		// idx points to the exact arg that isn't a flag. We increment
+		// by one so that all the copying below expects idx to be the
+		// insertion point.
+		idx++
+
+		// Copy the args
+		newArgs := make([]string, len(args)+len(extra))
```
Comment (jbardin) [inline main.go]: Do we need to worry about non-flag arguments in the `TF_CLI_ARGS` variable? ( I can't think of a case where it matters, just tossing out the question)

Comment (mitchellh) [inline main.go]: I think we just make explicit that we insert the args right after the command. If that works, great, if that doesn't for them for normal args, then its documented.

File: main.go (hunk 1 above)
Comment (mitchellh) [inline main.go]: Really good catch, really good. Fixing now.

**Pull request merged**